### DISCUSSION TITLE
[✨]Non-Routable URL Route Param

### DISCUSSION THREAD

**Discussion by mhevery** [QwikDev/qwik-evolution#274] [Proposals For Qwik]
### Is your feature request related to a problem?

1. Imagine I have my application which has a set of routes: `['/', '/[item]']`
2. Now, I would like to internationalize the application. The way I would like to do this is to add `/[lang]/` as a prefix to my routes. So the new routes are really `['/[lang]/', '/[lang]/[item]']`.

How can I achieve the above today? 
There are couple of issues we run into:
1. there is no easy way to create route prefixes
2. Within the application the best practice is to have absolute URLs. This creates an issue since the URL now contains a prefix.

### Describe the solution you'd like

When creating an application, I would like to:
1. Specify a prefix param that should be outside of my application routes.
2. Have URL building take the prefix into account.

### Describe alternatives you've considered

explicitly add `/[lang]` into the application route.

Issues:
- When building routes it may be hard to know the current `lang`
- navigation across `lang` should never be SPA (always MPA). Really navigation across `lang` is a different website.

**Comment by wmertens** [comment]
Would it be an option to use `<base href="/en" />` in the head?

**Comment by wmertens** [comment]
Another option: just use `$localize`

```
<a href={$localize`/blog/${slug}`}>Blog Post about {slug}</a>
```

Each locale gets to decide the translation, and the routes have to be able to accept those.
Meaning, the urls could be `/le-blog/${slug}` but also `/fr/blog/${slug}`. One or more layout.tsx files must have an onGet that uses the URL to decide the language.

I think this would just work?

To make it easier to write i18n routes, qwik-city could also accept comma-separated directory names. It can be useful for apps with only a few languages: `/src/routes/blog,le-blog,der-blog/layout.tsx`.

**Comment by mhevery** [comment]
> Would it be an option to use `<base href="/en" />` in the head?

That is an interesting idea, but how would it work? Mainly when you are creating `<a href="/foo"/>` is absolute, so how would you do relative? 🤔

**Comment by wmertens** [comment]
We could do relative paths with #146 

I implemented my other idea in https://github.com/mhevery/qwik-i18n/pull/11

**Comment by gioboa** [comment]
We moved this issue to `qwik-evolution` repo to create a RFC discussion for this.
Here is our [Qwik RFC process](https://github.com/QwikDev/qwik-evolution?tab=readme-ov-file#qwik-rfc-process) thanks.

**Comment by wmertens** [comment]
See #146 for the discussion on makePath, but I think the approach with just translating the URL using https://github.com/wmertens/compiled-i18n or $localize is fine (and Misko merged the PR implementing it), so I'll close this.

**Comment by crossterra-agent** [comment]
Created planner discussion: https://github.com/CrossTerra/monorepo/discussions/2035
